### PR TITLE
refactor: Convert the workspace-search plugin to Typescript

### PR DESCRIPTION
**Pull request by gonfunko**
This PR fixes part of #1647 by converting the workspace-search plugin to Typescript.

### CODE REVIEW

File: plugins/workspace-search/src/workspace_search.ts [hunk 1]
```diff
@@ -0,0 +1,581 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @fileoverview Class responsible for workspace search.
```
Comment (maribethb) [inline plugins/workspace-search/src/workspace_search.ts]: You can remove this part (fileoverview is optional, we removed authors from most places already)

File: plugins/workspace-search/src/workspace_search.ts [hunk 2]
```diff
@@ -0,0 +1,581 @@
...
+   * @returns The component’s bounding box. Null in this
+   *     case since we don't need other elements to avoid the workspace search
+   *     field.
+   */
+  getBoundingRectangle(): Blockly.utils.Rect|null {
+    return null;
+  }
+
+  /**
+   * Positions the zoom-to-fit control.
```
Comment (maribethb) [inline plugins/workspace-search/src/workspace_search.ts]: can you fix this text while you're here? it's wrong in both name and description

**Review by maribethb — APPROVED** [review]
There should be a devDependency on both `typescript` and `@typescript-eslint/parser` (see the `field-angle` plugin for an example; it's a mistake that the others don't list their dependency on ts. i can't find the issue where christopher and i debated about this though.)

File: plugins/workspace-search/src/workspace_search.ts (hunk 1 above)
Comment (gonfunko) [reply plugins/workspace-search/src/workspace_search.ts]: Done

File: plugins/workspace-search/src/workspace_search.ts (hunk 2 above)
Comment (gonfunko) [reply plugins/workspace-search/src/workspace_search.ts]: Done!

**Comment by gonfunko** [comment]
> There should be a devDependency on both `typescript` and `@typescript-eslint/parser` (see the `field-angle` plugin for an example; it's a mistake that the others don't list their dependency on ts. i can't find the issue where christopher and i debated about this though.)

Done!

**Pull request merged**